### ISSUE TITLE
[RSK][Ledger][Bug] Don't received confirmation window from Ledger

### ISSUE THREAD

**Issue by IntinteDAO** [poanetwork/nifty-wallet#375]
If you are reporting a problem with Nifty Wallet, please include the following information:

### What browser and version did you use?
Chromium 81.0.4044.92
### What operating system did you use?
Devuan Ceres (Debian Sid)
### Do you have screenshots showing the problem?
https://youtu.be/XHBiV-q9DDE
Ofc. I waiting for Ledger confirmation screen, but I've never received it

### What is your issue?
So I sent a transaction, but I don't received info from ledger with confirmation this TX.

**Comment by vbaranov** [comment]
@IntinteDAO You should first connect to Ledger before sending tx: Account menu -> Connect hardware wallet. A connection is lost every time you logout from the wallet. And as far as I know, there is a limited lifetime of connection to Ledger. So after some period of time working with the wallet even without logout Ledger disconnects.

**Comment by IntinteDAO** [comment]
Works, sorry for that. Maybe you can create some kind of auto-refresher? :-)

**Comment by vbaranov** [comment]
I don't think we can implement an auto-refresher because re-connection to Ledger implies user action to unlock it. The one possible way we can handle it is to show a warning at sending of tx from Ledger account if the hardware wallet is locked.

**Comment by IntinteDAO** [comment]
Okay, you can do it, though it's weird that you can't do it like I say.

I mean, what I had to do was go through the whole "Ledger's wallet connection" procedure with the unlocked Ledger. I think Nifty Wallet could go through that communication himself just in case.

**Comment by highplane** [comment]
[Bug] while using Ledger with nifty, sending and receiving RSK then switching to Ethereum from the menu in the top left corner the same address appears for both RSK and Ethereum.

**Comment by vbaranov** [comment]
>[Bug] while using Ledger with nifty, sending and receiving RSK then switching to Ethereum from the menu in the top left corner the same address appears for both RSK and Ethereum.

I guess this is related to this issue https://forum.poa.network/t/reveal-seed-words-derives-with-eth-dpath-instead-of-rsk/3263. We prepared a fix for this issue https://github.com/poanetwork/nifty-wallet/pull/356. And this one is a part of 5.1.0 release which has been sent to review several hours ago.

>I mean, what I had to do was go through the whole "Ledger with wallet connection" procedure with the unlocked Ledger. I think Nifty Wallet could go through that communication himself just in case.

I got your point. When I am saying "implies user action to unlock it" I mean the case when Ledger is locked which happens every *n* minutes. In that case, the auto refresher will not solve the issue. But I assume that cases exist when Ledger is still unlocked but Nifty loses connection. We will try to address those case in the future releases.

**Comment by jonathansmirnoff** [comment]
Hi! I would like to comment that are several user with a similar problem:
Nifty doesn't trigger the signal to Ledger to sign the tx, or the signal is too late that the transactions is rejected by the Nifty.

The workaround that I had do was the following:
1 - reset account
2 - disconnect ledger
3 - reconnect again.

The error is a bit random but several users had it.

I hope that you can review it. Thanks
